src/js/app.js

```diff
@@ -92,8 +78,53 @@ var toggle_popup_flag = (url) => {
 
 // TODO: do some hostname preprocessing before append
 // TOOD: write some logic for storing this in options and persist
-var blacklist = new Set(["www.facebook.com", "github.com"]);
-blacklist.forEach(toggle_popup_flag)
+
+// an event listener that calls update_blacklist when users want to update from content scripts
+chrome.runtime.onMessage.addListener( // how to receive an object and identify it LOL
+    (request,sender,sendResponse) => {
+        if (request.greeting == {K,V}) // TODO: this is tricky; how do you specify a format
+        update_blacklist({k:v});
+    } // call update_blacklist with the input received 
+    // input should be a KV: {domain: value}
+);
+
+
+// a typical pair is {"dom":UrlSettings Object}
+// need "dom" because we want to look up easily from chrome storage
+// TODO: chrome.runtime.sendMessage() 
+// here is another function that does update the keyvalue pair. done through sending messages to app.js
+
+chrome.runtime.onMessage.addListener(function(response,sender,sendResponse){
+    chrome.storage.sync.get(dom, function(object) {
```

Comment:
not sure what `dom` is referring to, since it's an undefined variable. If I'm understanding the use case of this listener correctly, you'd probably want to send in the `dom` (domain? host name?) as part of the request object (you name it `response` here for some reason..) e.g. with `chrome.runtime.sendMessage({dom: "facebook.com"}.....)` in the content script, and then access that message with `request.dom` in the background script here.